library/Terminal42/ChangeLanguage/FrontendModule/ChangeLanguageModule.php

```diff
@@ -158,6 +166,40 @@ protected function getCurrentPage()
     }
 
     /**
+     * @return array
+     */
+    protected function getRootPageIds()
+    {
+        $rootPageIds = explode(',', $this->rootPageIds);
+        $negateRootPageIds = (bool)$this->negateRootPageIds;
+        $currentPage = $this->getCurrentPage();
+
+        /**
+         * usecase: you want to show all rootPageIds which are NOT configured in the module config
+         * if the rootId of the current page is part of the list, we need to remove it - if its not removed then the
+         * active page will not be shown in the list
+         */
+        if ($negateRootPageIds && in_array($currentPage->rootId, $rootPageIds)) {
+            $key = array_search($currentPage->rootId, $rootPageIds);
+
+            if ($key !== false) {
+                unset($rootPageIds[$key]);
+            }
+        }
+
+        /**
+         * usecase: you want to show all rootPageIds which are configured in the module config
+         * if the rootId of the current page is NOT part of the list, we need to add it - if its not added then the
+         * active page will not be shown in the list
+         */
+        if (!$negateRootPageIds && !in_array($currentPage->rootId, $rootPageIds)) {
+            $rootPageIds[] = $currentPage->rootId;
+        }
```

Comment:
I've been thinking some more, maybe we should move both of these conditions into the `PageFinder` instead. But I am torn on this.

On one hand the way it's currently done modules can decide themselves how you want to handle the "current page" root (if you want it included / excluded based on the config).

On the other hand it maybe makes more sense to just pass rootIds `1,25,54` and then the PageFinder can add / remove a rootId in its own config  (based on the "current" Page that has been passed to `findRootPagesForPage` or `findMasterRootForPage`)

Reply:
I think you right. The page finder should be responsible for this

Reply:
see ad3a9ef